src/Expander.hs

```diff
@@ -1236,11 +1238,15 @@ expandOneForm prob stx
                   linkPattern dest $
                     CtorPattern ctor subPtrs
             other ->
-              throwError $ WrongMacroContext stx ExpressionCtx (problemContext other)
-        EPrimModuleMacro _ ->
-          throwError $ WrongMacroContext stx (problemContext prob) ModuleCtx
+              throwError $ WrongSyntacticCategory stx ExpressionCat (problemCategory other)
```

Comment:
Seeing as how these keep getting flipped WRT expected/actual, perhaps we should make newtypes for them? Something like `newtype Expected a = Expected a` and `newtype Actual a = Actual a`? Not necessarily in this PR, but it did seem useful to require it here.